packages/blocks/src/code-block/code-block.ts

```diff
@@ -84,19 +98,22 @@ export class CodeBlockComponent extends NonShadowLitElement {
             }}
           ></lang-list>
         </div>
-        <rich-text
-          .host=${this.host}
-          .model=${this.model}
-          .modules=${{
-            syntax: {
-              highlight: highlight.highlight,
-              codeBlockElement: this,
-              language: this.model.language,
-            },
-          }}
-        >
-          <div id="line-number"></div>
-        </rich-text>
+        ${this.highlight
+          ? html`<rich-text
+              .host=${this.host}
+              .model=${this.model}
+              .modules=${{
+                syntax: {
+                  highlight: this.highlight,
+                  codeBlockElement: this,
+                  language: this.model.language,
+                },
+              }}
+            >
+              <div id="line-number"></div>
+            </rich-text>`
+          : // TODO update loading state
+            `Loading coding block...`}
```

Comment:
The loading state should be handled by page block.
Should we have something like `Suspense` in React? /cc @doodlewind

Reply:
Can we use the [async rendering](https://lit.dev/docs/templates/directives/#asynchronous-rendering) directives in Lit?

Reply:
wow this is great

Reply:
The until directive will cause the UI to always have to wait for a trick because it returns nothing if the argument is not a promise.

So I will keep the existing code style.